### PR TITLE
Add DragDropContextProvider to types/react-dnd/index.d.ts

### PR DESCRIPTION
**Pull request by chrisdoble**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://react-dnd.github.io/react-dnd/docs-drag-drop-context-provider.html

### CODE REVIEW

File: types/react-dnd/index.d.ts [hunk 1]
```diff
@@ -52,6 +52,9 @@ declare module __ReactDnd {
         backend: Backend
     ): <P>(componentClass: React.ComponentClass<P> | React.StatelessComponent<P>) => ContextComponentClass<P>;
 
+    export class DragDropContextProvider extends React.Component<{ backend: Backend }> {
```
Comment (vsiao) [inline types/react-dnd/index.d.ts]: can you pull this out into a separate `DragDropContextProviderProps` interface and also add the optional `window` prop?

**Review by vsiao — COMMENTED** [review]
Thanks for your contribution. Can you add a test for this? The example given in the documentation will suffice.

**Comment by aozgaa** [comment]
Waiting on revision from @chrisdoble.

**Comment by chrisdoble** [comment]
Closing because we no longer need these types in our project and it's unlikely I'll have time to revise this in the near future. Others are welcome to take over.

**Comment by joscha** [comment]
fixed in #19765

**Pull request closed without merging**